modules/reactor/test/tests/meshgenerators/reporting_id/hexagonal_id/tests

```diff
@@ -0,0 +1,48 @@
+[Tests]
+  design = 'meshgenerators/HexIDPatternedMeshGenerator.md'
+  issues = '#19217'
+  [assembly]
+    requirement = 'The system shall support the generation of reporting IDs for hexagonal assemblies'
+    [cell]
+      type = 'Exodiff'
+      input = 'assembly_reporting_id.i'
+      exodiff = 'assembly_reporting_id_cell_out.e'
+      cli_args = "Outputs/file_base='assembly_reporting_id_cell_out'"
+      detail = 'assign pin IDs in sequential order for each components in the lattice'
+      recover = false
+    []
+    [pattern]
+      type = 'Exodiff'
+      input = 'assembly_reporting_id.i'
+      exodiff = 'assembly_reporting_id_pattern_out.e'
+      cli_args = "Mesh/assembly/assign_type='pattern' Outputs/file_base='assembly_reporting_id_pattern_out'"
+      detail = 'assign pin IDs based on the ID of the input pattern tiles'
+      recover = false
+    []
+    [manual]
+      type = 'Exodiff'
+      input = 'assembly_reporting_id.i'
+      exodiff = 'assembly_reporting_id_manual_out.e'
+      cli_args = "Mesh/assembly/assign_type='manual' Mesh/assembly/id_pattern='2 2 2;2 1 1 2;2 1 0 1 2;2 1 1 2;2 2 2' Outputs/file_base='assembly_reporting_id_manual_out'"
+      detail = 'assign pin IDs based on user-defined mapping'
+      recover = false
+    []
+  []
+  [core]
```

Comment:
I think these can probably be split up with just two requirements and detail, but I don't completely follow what they're testing